TwilioVerifySDKTests/TwilioSecurity/Sources/Storage/AuthenticatedSecureStorageTests.swift

```diff
@@ -0,0 +1,242 @@
+//
+//  AuthenticatedSecureStorageTests.swift
+//  TwilioVerifySDKTests
+//
+//  Copyright © 2021 Twilio.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+import LocalAuthentication
+@testable import TwilioVerifySDK
+
+class AuthenticatedSecureStorageTests: XCTestCase {
+
+  var testSubject: AuthenticatedSecureStorage!
+  var keychainMock: KeychainMock!
+
+  private enum MockErrors: Error {
+    case unexpected
+  }
+
+  override func setUp() {
+    keychainMock = .init()
+    testSubject = .init(keychain: keychainMock, keychainQuery: KeychainQuery())
+  }
+
+  // MARK: - Save Data
+
+  func testSave_withValidValuesSavingData_shouldNotThrow() {
+    // Given
+    let data = "testData".data(using: .utf8)
+    let key = "testKey"
+    let authenticator = AuthenticatorMock(context: MockContext())
+    keychainMock.addItemStatus = [errSecSuccess]
+
+    // When
+    testSubject.save(data!, withKey: key, authenticator: authenticator) { [self] in
+      // Then
+      XCTAssertEqual(self.keychainMock.callsToAddItem, 1)
+      XCTAssertEqual(self.keychainMock.callOrder.first, .addItem)
+    } failure: { error in
+      XCTFail("Should not retrieve error: \(error.localizedDescription)")
+    }
+  }
+
+  func testSave_withValidValuesSavingDataButNotKeyChainAvailable_shouldThrow() {
+    // Given
+    let data = "testData".data(using: .utf8)
```

Comment:
Is there any reason to create the mock data, key and authentication on every test? Looks like you are using the same data for all of them, maybe I'm wrong though

Reply:
IMO Creating mock data shared in the entire test class is a bad practice, if for some reason you need modify it further it would make everything hard to maintain, you could end up modifying it for all the tests. I prefer creating Enums with mock data, if the test data is enough large that may occupy too much lines of code per test, and will be used multiple times. In this case, I think is no needed.

Reply:
Sounds fair :)